src/hypothesis/strategies.py

```diff
@@ -1064,18 +1064,16 @@ def decimals(min_value=None, max_value=None,
     # Set up a strategy for finite decimals
     if places is not None:
         # Fixed-point decimals are basically integers with a scale factor
-
-        def try_quantize(d):
-            try:
-                return d.quantize(factor)
-            except InvalidOperation:  # pragma: no cover
-                return None
+        # plus complicated issues about context and precision
         factor = Decimal(10) ** -places
-        max_num = max_value / factor if max_value is not None else None
-        min_num = min_value / factor if min_value is not None else None
-        strat = integers(min_value=min_num, max_value=max_num)\
-            .map(lambda d: try_quantize(d * factor))\
-            .filter(lambda d: d is not None)
+        default = Decimal(10) ** (getcontext().prec - places) - factor
```

Comment:
After spending some time staring at this yesterday and mulling on it overnight, I'm afraid I think this is the wrong solution.

The big direct problem with it is that it interacts wrongly with laziness and caching. What will happen here is that we'll build the default for all decimals strategies (with the same arguments) off the decimal context wherever we happen to have first run a test/example/whatever with it. In particular *not* where we first defined it, and certainly not where we defined it the second time even if there happened to be a different decimal context there.

In general I think we should try to achieve a consistent set of behaviours for strategies regardless of the external state, so I think the right thing to do here is to create and use an appropriate decimal context rather than inheriting the current one.